apis/actions.github.com/v1alpha1/autoscalingrunnerset_types.go

```diff
@@ -80,6 +85,124 @@ type ProxyConfig struct {
 
 	// +optional
 	HTTPS *ProxyServerConfig `json:"https,omitempty"`
+
+	// +optional
+	NoProxy []string `json:"noProxy,omitempty"`
+}
+
+func (c *ProxyConfig) ProxyFunc(secretFetcher func(string) (*corev1.Secret, error)) (func(*http.Request) (*url.URL, error), error) {
```

Comment:
feels like these methods should live in some http client.go files instead of the types.go

Reply:
We can type the function returned here in client, that's a good shout. This feels like a method for ProxyConfig though, it knows everything it needs about itself already and personally it doesn't feel right to couple a client with a k8s type. What do you think?

Reply:
Also worth noting the returned func isn't a client specific detail but it's the type [http.Transport](https://pkg.go.dev/net/http#Transport) expects